android/app/src/main/java/app/organicmaps/MwmActivity.java

```diff
@@ -2004,6 +2064,61 @@ public void onSearchRoutePoint(@RoutePointInfo.RouteMarkType int pointType)
   @Override
   public void onRoutingStart()
   {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M)
+    {
+      if (!mNavigationBatterySaverPermission &&
+          !mPowerManager.isIgnoringBatteryOptimizations(getPackageName()))
+      {
+        Logger.w(TAG, "Checking for App battery optimisation permission");
+        mNavigationBatterySaverPermission = true;
+        dismissAlertDialog();
+        final MaterialAlertDialogBuilder builder = new MaterialAlertDialogBuilder(this, R.style.MwmTheme_AlertDialog)
```

Comment:
1) if user denies, in which condition we gonna again ask this permission?

2) if user allowed and after navigation started he reverted the permissions. Will we be asking him instantly or on every new navigation start?

Reply:
@kavikhalique How do other apps behave in such situations?

Reply:
I tested starva and mapmyride, they both 

1) For battery optimisation and battery saver they asked only on first start after that they didn't asked even if the feature is turned on.

2) For location permission they ask every time if it is disabled

3) For post notification permission i am not sure because i do not have a device which have android 13 : )

But here the issue is if power saving mode is turned on we don't get satellite fix that means no location update at all which makes trace path feature useless. 

I do not have proper knowledge how these apps deal with this problem in power saving mode but probably since they are not offline apps they may acquire location from other sources (network, wifi etc). But this might not be case for us since many users might be using it because of its total offline functionality.